sigstore/_internal/rekor/_client.py

```diff
@@ -2,8 +2,83 @@
 Client implementation for interacting with Rekor.
 """
 
+import json
+from urllib.parse import urljoin
+
+import requests
+
+DEFAULT_REKOR_URL = "https://rekor.sigstore.dev/api/v1/"
+
+
+class Endpoint:
+    def __init__(self, url, session):
+        self.url = url
+        self.session = session
+
+    def get(self):
+        raise NotImplementedError
+
+    def post(self):
+        raise NotImplementedError
+
 
 class RekorClient:
     """The internal Rekor client"""
 
-    pass
+    def __init__(self, url=DEFAULT_REKOR_URL):
+        self.url = url
+        self.session = requests.Session()
+        self.session.headers.update(
+            {"Content-Type": "application/json", "Accept": "application/json"}
+        )
+
+    @property
```

Comment:
Nice idea with this `Endpoint` pattern. Fulcio has fewer endpoints but I think I'll use that pattern there too.